### PR TITLE
Generating Handwritten Digits on MNIST using GANs

### PR DESCRIPTION
**Pull request by swaingotnochill**
This PR is for discussing the issues and working on GANs notebook for MNIST dataset.

### CODE REVIEW

**Comment by swaingotnochill** [comment]
@zoq @kartikdutt18  When I am adding convulation layer, it seems to give an error that 

> /home/viole/anaconda3/envs/notebook/include/mlpack/methods/ann/ffn.hpp:290:36: error: no member named 'push_back' 
Is it due to syntax error while using the modules or something else? I tried working around it, but don't know why its giving push_back not found(this function should be available).

your thoughts?

File: generating_hand_written_digits_mnist_with_gan/generating_handwritten_digits_mnist_with_gan.ipynb [hunk 1]
```diff
@@ -0,0 +1,404 @@
+{
```
Comment (kartikdutt18) [inline generating_hand_written_digits_mnist_with_gan/generating_handwritten_digits_mnist_with_gan.ipynb:1]: <pre class="ql-syntax" spellcheck="false">model.Add(new mlpack::ann::Convolution&lt;&gt;(,6,5,5,1,1,0,0,28,28))
</pre><p>Could you try this?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlpack/examples/pull/161/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (swaingotnochill) [reply generating_hand_written_digits_mnist_with_gan/generating_handwritten_digits_mnist_with_gan.ipynb:1]: I tried but it doesn't seems to work too.
`input_line_65:2:16: error: no matching constructor for initialization of 'mlpack::ann::Convolution<>'
 model.Add(new mlpack::ann::Convolution<>(1,6,5,5,1,1,0,0,28,28));
               ^                          ~~~~~~~~~~~~~~~~~~~~~
/home/viole/anaconda3/envs/notebook/include/mlpack/methods/ann/layer/layer_types.hpp:172:7: note: candidate constructor (the implicit copy constructor) not viable: requires
      1 argument, but 10 were provided
class Convolution;
      ^
/home/viole/anaconda3/envs/notebook/include/mlpack/methods/ann/layer/layer_types.hpp:172:7: note: candidate constructor (the implicit move constructor) not viable: requires
      1 argument, but 10 were provided`

No matching constructor..

Comment (zoq) [reply generating_hand_written_digits_mnist_with_gan/generating_handwritten_digits_mnist_with_gan.ipynb:1]: I just tested it on lab.mlpack.org and there it works fine, I had some problems with namespaces in the past, can you put each namespace into a separate cell?

**Review by zoq — COMMENTED** [review]
For an example about how to save/load a model you can checkout https://github.com/mlpack/examples/blob/master/mnist_simple/mnist_simple.cpp, the interesting part is:

https://github.com/mlpack/examples/blob/master/mnist_simple/mnist_simple.cpp#L170

and 

https://github.com/mlpack/examples/blob/776e116140ae2c4ed61a290ea8f1fee10fa174c0/mnist_simple/mnist_simple.cpp#L174

Let me know if I should clarify anything further.

**Comment by kartikdutt18** [comment]
Hi Swain, Could you push the latest changes (if you have got the issue resolved).

**Comment by swaingotnochill** [comment]
> Hi Swain, Could you push the latest changes (if you have got the issue resolved).

Sure...I will push it by Tonight (I don't have the computer right now )

**Comment by swaingotnochill** [comment]
I redirected the PR at #172 . Closing this one.

**Pull request closed without merging**